lib/smart_answer_flows/maternity-paternity-calculator/maternity_calculator_flow.rb

```diff
@@ -68,21 +50,16 @@ def define
         radio :did_the_employee_work_for_you_between? do
           option :yes
           option :no
-          calculate :not_entitled_to_pay_reason do |response|
-            response == "no" ? :not_worked_long_enough_and_not_on_payroll : nil
-          end
-          calculate :to_saturday do
-            calculator.qualifying_week.last
-          end
 
-          calculate :to_saturday_formatted do
-            calculator.format_date_day to_saturday
+          on_response do |response|
+            self.has_employment_contract_between_dates = response
+            self.not_entitled_to_pay_reason = response == "no" ? :not_worked_long_enough_and_not_on_payroll : nil
```

Comment:
That was my original solution, but it caused the tests to fail. The problem being that when `response == "no"` is false, `not_entitled_to_pay_reason` doesn't get assigned. State is an OpenStruct, so when not assigned the method doesn't exist on the object.

I changed it to the tertiary version so that when false, `not_entitled_to_pay_reason` is assigned `nil`.